### DISCUSSION TITLE
observations frequency change attribute error

### DISCUSSION THREAD

**Discussion by tdmeij** [pastas/pastas#199] [Q&A] [answered]
For a model that showed autocorrelation in the diagostics() plot I tried to change the observations frequency from daily values to weekly observations using the expression: obs.update_series(freq='W ',sample_down='mean'). This gives an Attribute error: 'MonthEnd' object has no attribute 'delta'.

After some searching in the Pastas code it appears that the Note section in the function utils.frequency_is_supported(freq) states that monthly ('M '), yearly (Ý') or  weekly ('W') frequencies are not allowed for technical reasons because they are not equidistant. 

It would be helpful if the obs.update_series() method showed a user warning in case a user tries to change the frequency to a valid pandas frequency that is not supported. Also some instructions in the help section on how to downsample an observation series to weekly, monthly or yearly values would be greatly appreciated.

**Comment by raoulcollenteur** [comment, marked as answer]
Hi Thomas,

Thanks for reporting the Issue! I would recommend not to use Pastas methods for this purpose, and instead resample the original Pandas Series. This way you have a better control of what is happening with the time series.

We have not yet really experimented with using the Pastas methods for up or downsampling **dependent time series**, in particular when dealing with irregular time steps (so the original frequency cannot be automatically determined). For example:

`ml.oseries.update_series(freq='14D', sample_down="mean")`

Will not do that what you expect when irregular time steps are present (see also #68), and time_step_weighted_resample will be used. It works only when regular time steps are provided now.

Cheers,
Raoul

**Comment by tdmeij** [comment]
Hi Raoul

I've been experimenting with downsampling a Pastas observation series from
days to months or weeks. Sampling to Months seems to work just fine, but
resampling to Weeks gives another issue. When I do:
>>>precfile = 'neerslaggeg_WEERSELO_675.txt'
>>>prec = ps.read_knmi(precfile, variables='RD')
>>>precW = prec.series_originall.resample('W').sum()
Pastas gives this error:
>>>ValueError: freq of W-SUN not supported

In pastas module utils.py I find on line number 96 this line:   e
   elif freq in ['W']:
When I change this to:
   elif freq in ['W','W-SUN']:
Pastas works as expected.

I tried to avoid the issue by resampling the time series with
.resample('7D').sum() but Pastas figures out that this must be a weekly
frequency and changes '7D' to 'W-SUN'.:)

I'am not sure if I should report this as an issue in Pastas on Github or
post a question on Stackoverflow. I've been searching for ways to resample
a Pandas Time series to weeks with a frequency 'W' as a result, but the
added day seems to be a logical necessity of converting to weeks.

greetings, Thomas


Op do 28 mei 2020 om 17:18 schreef Raoul Collenteur <
notifications@github.com>:

**Reply by dbrakenhoff** [reply]
This Pandas thing where it defaults to e.g. W-SUN instead of W or 7D is a major pain, and is something we might avoid when we review the TimeSeries object for Pastas 1.0, or when I take a better look at timeseries models with calculation freq >1D.
